plugin/pkg/scheduler/generic_scheduler.go

```diff
@@ -125,6 +126,10 @@ func findNodesThatFit(pod *api.Pod, podLister algorithm.PodLister, predicateFunc
 				if _, found := failedPredicateMap[node.Name]; !found {
 					failedPredicateMap[node.Name] = util.StringSet{}
 				}
+				if predicates.FailedResourceType != "" {
+					failedPredicateMap[node.Name].Insert(predicates.FailedResourceType)
```

Comment:
FailedResourceType is given value at https://github.com/HaiyangDING/kubernetes/blob/ImproveClarityResource/plugin/pkg/scheduler/algorithm/predicates/predicates.go#L168 (and line 163/173 for other 2 cases respectively). Since `FailedResourceType`  can be called from outside package predicates, so I just use it here.  Every time before using it, `FailedResourceType`  is set to '""' at https://github.com/HaiyangDING/kubernetes/blob/ImproveClarityResource/plugin/pkg/scheduler/generic_scheduler.go#L119.